core/src/main/scala/higherkindness/mu/rpc/srcgen/compendium/CompendiumClient.scala

```diff
@@ -28,62 +20,36 @@ trait CompendiumClient[F[_]] {
 
 object CompendiumClient {
 
-  def apply[F[_]]()(implicit
-      interp: InterpTrans[F],
-      clientConfig: HttpConfig,
-      F: Sync[F]
+  def apply[F[_]: Sync](
+      clientF: Client[F],
+      clientConfig: HttpConfig
   ): CompendiumClient[F] =
     new CompendiumClient[F] {
 
       val baseUrl: String          = s"http://${clientConfig.host}:${clientConfig.port}"
       val versionParamName: String = "version"
 
-      override def storeProtocol(identifier: String, protocol: RawProtocol): F[Int] = {
-        val request: Free[HttpF, HttpResponse] =
-          Hammock.request(Method.POST, uri"$baseUrl/v0/protocol/$identifier", Map(), Some(protocol))
-
-        for {
-          status <- request.map(_.status).exec[F]
-          _ <- status match {
-            case Status.Created => Sync[F].unit
-            case Status.OK      => Sync[F].unit
-            case Status.BadRequest =>
-              asError(request, SchemaError)
-            case Status.InternalServerError =>
-              Sync[F].raiseError(UnknownError(s"Error in compendium server"))
-            case _ =>
-              Sync[F].raiseError(UnknownError(s"Unknown error with status code $status"))
-          }
-        } yield status.code
-      }
-
       override def retrieveProtocol(
           identifier: String,
           version: Option[Int]
       ): F[Option[RawProtocol]] = {
         val versionParam = version.fold("")(v => s"?$versionParamName=${v.show}")
-        val uri          = uri"$baseUrl/v0/protocol/$identifier$versionParam"
+        val uri          = s"$baseUrl/v0/protocol/$identifier$versionParam"
 
-        val request: Free[HttpF, HttpResponse] = Hammock.request(Method.GET, uri, Map())
+        implicit val rawEntityDecoder = jsonOf[F, RawProtocol]
 
-        for {
-          status <- request.map(_.status).exec[F]
-          out <- status match {
-            case Status.OK       => request.as[RawProtocol].map(Option(_)).exec[F]
+        clientF.get(uri)(res =>
+          res.status match {
+            case Status.Ok       => res.as[RawProtocol].map(Option(_))
             case Status.NotFound => Sync[F].pure(None)
```

Comment:
**question**: If something is not found, it worth to notify the user?

Reply:
With the current development, it raises an error: 
```
case None =>
              Async[F].raiseError[File](
                ProtocolNotFound(s"Protocol ${protocolAndVersion.name} not found in Compendium. ")
              )
```
on `run` function

Reply:
I think it makes sense to just return `None` at this point, because we are just an HTTP client and we don't know what we are being used for. It's pretty reasonable to try to get a resource which might not exist, so it's not an exceptional circumstance.

Whoever is using the client can decide whether a missing resource should be an error or not.